snap/validate_test.go

```diff
@@ -564,6 +564,36 @@ func (s *ValidateSuite) TestValidateLayout(c *C) {
 	c.Check(ValidateLayout(&Layout{Snap: si, Path: "$SNAP/data", Symlink: "$SNAP_DATA"}, nil), IsNil)
 }
 
+func (s *ValidateSuite) TestValidateLayoutAll(c *C) {
+	// /foo prevents /foo/bar from being valid (tmpfs)
+	const yaml1 = `
+name: broken-layout-1
+layout:
+  /usr/foo:
+    type: tmpfs
+  /usr/foo/bar:
```

Comment:
Can a test case where the order of `/usr/foo/bar` and `/usr/foo` is swapped?

Reply:
This is effectively it since dictionaries are not ordered, right? I'll add something quickly to be sure.